### PR TITLE
New data source: aws_ec2_transit_gateway_dx_gateway_attachment

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/8590

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
FEATURES:

* New Data Source: aws_ec2_transit_gateway_dx_gateway_attachment (#8590)
```

_WIP_ as this PR requires https://github.com/terraform-providers/terraform-provider-aws/pull/8528 to be able to run the acceptance tests.

### CODE REVIEW

**Comment by ewbankkit** [comment]
Rebased after merging of https://github.com/terraform-providers/terraform-provider-aws/pull/8528.

Acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -parallel 20 -run=TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_ -timeout 120m
=== RUN   TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_TransitGatewayIdAndDxGatewayId
=== PAUSE TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_TransitGatewayIdAndDxGatewayId
=== CONT  TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_TransitGatewayIdAndDxGatewayId
--- PASS: TestAccAWSEc2TransitGatewayDxGatewayAttachmentDataSource_TransitGatewayIdAndDxGatewayId (1297.02s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1297.042s
```

Removed _WIP_.

**Comment by bflad** [comment]
This has been released in [version 2.14.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2140-june-06-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

File: aws/data_source_aws_ec2_transit_gateway_dx_gateway_attachment.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	input := &ec2.DescribeTransitGatewayAttachmentsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("resource-id"),
+				Values: []*string{aws.String(d.Get("dx_gateway_id").(string))},
+			},
+			{
+				Name:   aws.String("resource-type"),
+				Values: []*string{aws.String("direct-connect-gateway")}, // Not yet defined in ec2/api.go.
```
[the remark this answers is not in the document]
Comment (ewbankkit) [reply aws/data_source_aws_ec2_transit_gateway_dx_gateway_attachment.go]: `ec2.TransitGatewayAttachmentResourceTypeDirectConnectGateway` has now been added in [AWS SDK v1.19.46](https://github.com/aws/aws-sdk-go/releases/tag/v1.19.46), merged via https://github.com/terraform-providers/terraform-provider-aws/pull/8864.

[the remark this answers is not in the document]
Comment (n3ph) [reply aws/data_source_aws_ec2_transit_gateway_dx_gateway_attachment.go]: I forgot to check the Go SDK... Right

**Pull request merged**